spot-contracts/contracts/RolloverVault.sol

```diff
@@ -125,51 +108,98 @@ contract RolloverVault is
     /// @notice The perpetual token on which rollovers are performed.
     IPerpetualTranche public perp;
 
+    //--------------------------------------------------------------------------
+    // v2.0.0 STORAGE ADDITION
+
+    /// @notice External contract that controls fees for minting, redeeming, swapping and deployment.
+    IFeePolicy public feePolicy;
+
+    /// @notice Reference to the wallet or contract that has the ability to pause/unpause operations.
+    /// @dev The keeper is meant for time-sensitive operations, and may be different from the owner address.
+    /// @return The address of the keeper.
+    address public keeper;
+
+    /// @notice The enforced minimum balance of underlying tokens to be held by the vault at all times.
+    /// @dev On deployment only the delta greater than this balance is deployed.
+    uint256 public minUnderlyingBal;
+
+    //--------------------------------------------------------------------------
+    // Modifiers
+
+    /// @dev Throws if called by any account other than the keeper.
+    modifier onlyKeeper() {
+        if (_msgSender() != keeper) {
+            revert UnauthorizedCall();
+        }
+        _;
+    }
+
     //--------------------------------------------------------------------------
     // Construction & Initialization
 
     /// @notice Contract state initialization.
     /// @param name ERC-20 Name of the vault token.
     /// @param symbol ERC-20 Symbol of the vault token.
     /// @param perp_ ERC-20 address of the perpetual tranche rolled over.
+    /// @param feePolicy_ Address of the fee policy contract.
     function init(
         string memory name,
         string memory symbol,
-        IPerpetualTranche perp_
+        IPerpetualTranche perp_,
+        IFeePolicy feePolicy_
     ) public initializer {
+        // initialize dependencies
         __ERC20_init(name, symbol);
         __ERC20Burnable_init();
         __Ownable_init();
         __Pausable_init();
         __ReentrancyGuard_init();
 
+        // set keeper reference
+        keeper = owner();
+
+        // setup underlying collateral
         underlying = perp_.collateral();
         _syncAsset(underlying);
 
+        // set reference to perp
         perp = perp_;
-    }
 
-    //--------------------------------------------------------------------------
-    // ADMIN only methods
+        // set the reference to the fee policy
+        updateFeePolicy(feePolicy_);
 
-    /// @notice Pauses deposits, withdrawals and vault operations.
-    /// @dev NOTE: ERC-20 functions, like transfers will always remain operational.
-    function pause() external onlyOwner {
-        _pause();
+        // setting initial parameter values
+        minDeploymentAmt = 0;
```

Comment:
Since uninitialized contract values are 0, think we need these?

Reply:
Not really, but prefer keeping it in there for clarity